handlers/delete_lease.go

```diff
@@ -44,6 +46,18 @@ func DeleteLease(services k8s.ServiceGetterUpdater, k8sServiceName string) http.
 			htp.Error(w, http.StatusInternalServerError, "error saving new annotations (%s)", err)
 			return
 		}
+
+		// clear all namespaces
+		namespacesList, err := namespaces.List(api.ListOptions{LabelSelector: labels.Everything()})
+		if err != nil {
+			htp.Error(w, http.StatusInternalServerError, "cannot get namespaces (%s)", err)
+			return
+		}
+		for _, namespace := range namespacesList.Items {
+			if namespace.Namespace != "kube-system" && namespace.Namespace != "default" {
+				namespaces.Delete(namespace.Namespace)
```

Comment:
might be better to do each delete concurrently, but that doesn't have to block this PR. can you create an issue for it?